internal/ingress/controller/template/template.go

```diff
@@ -588,6 +595,33 @@ func buildAuthResponseHeaders(proxySetHeader string, headers []string) []string
 	return res
 }
 
+func buildAuthUpstreamHeaders(proxySetHeader string, headers []string) []string {
```

Comment:
Please add a comment on this function on why this is used, and what's the final result in nginx.conf file